rs/bitcoin/ckbtc/minter/src/state.rs

```diff
@@ -451,24 +454,29 @@ impl CkBtcMinterState {
         self.ledger_id = ledger_id;
         self.max_time_in_queue_nanos = max_time_in_queue_nanos;
         self.mode = mode;
-        self.kyt_principal = kyt_principal;
-        if let Some(kyt_fee) = kyt_fee {
-            self.kyt_fee = kyt_fee;
+        self.btc_checker_principal = btc_checker_principal;
+        if let Some(check_fee) = check_fee {
+            self.check_fee = check_fee;
+        } else if let Some(kyt_fee) = kyt_fee {
+            self.check_fee = kyt_fee;
         }
         if let Some(min_confirmations) = min_confirmations {
             self.min_confirmations = min_confirmations;
         }
     }
 
+    #[allow(deprecated)]
     pub fn upgrade(
         &mut self,
         UpgradeArgs {
             retrieve_btc_min_amount,
             max_time_in_queue_nanos,
             min_confirmations,
             mode,
+            check_fee,
+            btc_checker_principal,
+            kyt_principal: _,
```

Comment:
Actually `kyt_fee` and `kyt_principal` are only useful for replaying old events. It turns out that the replay does not dependent on having the `kyt_principal` remembered by the state because those events only need and already have `kyt_provider`. So it is okay to leave it out.